src/advanced_inv.cpp

```diff
@@ -1311,16 +1321,31 @@ bool advanced_inventory::action_move_item( advanced_inv_listitem *sitem,
                     player_character.activity.str_values.push_back( "force_ground" );
                 }
 
-                player_character.activity.targets.push_back( item_location( player_character,
-                        sitem->items.front() ) );
-                player_character.activity.values.push_back( amount_to_move );
+                int remaining_amount = amount_to_move;
+                for( auto it = sitem->items.begin();
+                     remaining_amount > 0 && it != sitem->items.end();
+                     ++it ) {
+                    player_character.activity.targets.emplace_back( player_character, *it );
+                    const int move_amount = ( *it )->count_by_charges() ?
+                                            std::min( remaining_amount, ( *it )->charges ) : 1;
+                    player_character.activity.values.emplace_back( move_amount );
+                    remaining_amount -= move_amount;
+                }
```

Comment:
I recommend a range-based for loop here.
```suggestion
                for( item *itm : sitem->items ) {
                    if( remaining_amount <= 0 ) {
                        break;
                    }
                    player_character.activity.targets.emplace_back( player_character, itm );
                    const int move_amount = itm->count_by_charges() ?
                                            std::min( remaining_amount, it->charges ) : 1;
                    player_character.activity.values.emplace_back( move_amount );
                    remaining_amount -= move_amount;
                }
```